### PR TITLE
Webhook tweaks

### PR DESCRIPTION
**Pull request by jeffmcaffer**
This is a modest set of tweaks to better enable adding on WebHook driven data gathering.

* Schema update -- Add email and name back to the schema def.  These are needed for folks setting up their own system and do not hurt anyone consuming dumps
* Org filtering -- when running your own ghtorrent it is common to hook only your own orgs.  However ,those transitively reach other orgs through users etc.  Added a filter list (orgs.txt) that, if supplied, limits the set of org to be considererd
* Rate limit boost -- Some people have API keys with higher rate limits.  Flip the meaning of req_limit to not depend on the capacity of the key as the app will never know that.

These changes are not fully tested but I wanted to get some early feedback on the content and form.

### CODE REVIEW

**Comment by jeffmcaffer** [comment]
For the record, these changes include work done by  @ryanfarr01 but he didn't have time to propose the changes before finishing his internship.

**Comment by jeffmcaffer** [comment]
Also added option to use SSL for connecting to Mongo.  This is needed when mongo is on a separate machine (e.g., DocumentDB in Azure :-)

**Comment by jeffmcaffer** [comment]
Added force fix.  Original code was testing :force_given but the option is really just :force

**Comment by jeffmcaffer** [comment]
Changed full repo retrieval to handle empty repos.  Previously pulling bits of an empty repo resulted in a 409 that would then error out of the retrieval stage loop.  While it is odd that it returns a 409 (Conflict?), it is not fatal nor really an error so we can continue and get issues etc.

File: sql/schema.sql [hunk 1]
```diff
@@ -15,7 +15,9 @@ DROP TABLE IF EXISTS `ghtorrent`.`users` ;
 CREATE TABLE IF NOT EXISTS `ghtorrent`.`users` (
   `id` INT(11) NOT NULL AUTO_INCREMENT COMMENT '',
   `login` VARCHAR(255) NOT NULL COMMENT '',
+  `name` VARCHAR(255) NULL DEFAULT NULL COMMENT '',
```
Comment (gousiosg) [inline sql/schema.sql]: This (and the `email`) are not fields that are distributed with the default MySQL dump (which is what this file is meant to support), due to ghtorrent/ghtorrent.org#32

File: lib/ghtorrent/adapters/mongo_persister.rb [hunk 2]
```diff
@@ -55,7 +55,18 @@ def initialize(set)
 
     def store(entity, data = {})
       super
-      mongo[entity].insert_one(data).to_s
+      begin
+        mongo[entity].insert_one(data).to_s
+      rescue StandardError => e
+        # In the interest of best effort, catch errors where the document is too big for Mongo
+        # Not great approach but the exception from Mongo is pretty generic.
+        if e.message.include?("Request size is too large")
```
Comment (gousiosg) [inline lib/ghtorrent/adapters/mongo_persister.rb]: I don't think I 've ever seen this. Is it perhaps a MongoDB misconfiguration on your side? In any case, this is too fragile to include in the code base.

Comment (gousiosg) [inline lib/ghtorrent/adapters/mongo_persister.rb]: Oops, I just shaw your comment above. So DocumentDB does not support docs bigger than 1MB?! https://azure.microsoft.com/en-us/documentation/articles/documentdb-limits/

**Comment by gousiosg** [comment]
Most changes look reasonable. 

The organization filter code is a bit too use-case-specific to include in `GHTorrent::Mirror`. One way to go ahead with it would be to create a subclass of  `GHTorrent::TransactedGHTorrent`, override/include the code you have now included in `GHTorrent::Mirror` and provide a command-line option to `GHTorrent::Commands::FullRepoRetriever` to load your subclass instead of the default `TransactedGHTorrent`.

I would be happy to include this change.

File: sql/schema.sql (hunk 1 above)
Comment (jeffmcaffer) [inline sql/schema.sql]: Right.  The challenge however is that someone setting up their own GHT will need to define the schema in their database.  That instance will have email and name fields. The code fails if they are not in the schema. We actually want these values as we're watching our own repos so the values are quite reasonable to harvest and we'll take care to use them properly.

The thinking here is thta it is ok to define columns that are empty all the time in some data sets so having them in the schema is not a particular problem.

File: lib/ghtorrent/adapters/mongo_persister.rb (hunk 2 above)
Comment (jeffmcaffer) [inline lib/ghtorrent/adapters/mongo_persister.rb]: it does but that needs to be changed by admins.  I'm getting that fixed and we should be ok.  This change will not be in the final PR.

File: lib/ghtorrent/command.rb [hunk 3]
```diff
@@ -94,7 +94,7 @@ def process_options
             :type => String
         opt :token, 'GitHub OAuth token',
             :type => String, :short => 't'
-        opt :req_limit, 'Request limit for provided account (in reqs/hour)',
+        opt :req_limit, 'Number or requests to leave on any provided account (in reqs/hour)',
```
Comment (jeffmcaffer) [inline lib/ghtorrent/command.rb]: How do you feel about this re-orientation of req_limit?  The idea is that we want to ensure a certain amount is left,  That eliminates the need to know how many are on any given key (we have a mix of big and normal keys).  It does however change a user input.  We could do a separate option (name?) and have them be mutually exclusive.

File: lib/ghtorrent/ghtorrent.rb [hunk 4]
```diff
@@ -54,6 +55,25 @@ def persister
...
+        x = x.strip
+        if x.empty? == false
+          result.add(x)
+        end
+      end
+      result
+    end
+
+    def includeOrg?(org)
+      return (not @orgFilter.nil?) || (@org_filter.include? org)
```
Comment (jeffmcaffer) [inline lib/ghtorrent/ghtorrent.rb]: The naming here was bogs as was some of the logic.  There is a revised version coming.

File: sql/schema.sql (hunk 1 above)
Comment (gousiosg) [inline sql/schema.sql]: The schema is automatically constructed the first time GHTorrent connects to the database. See the migration code under `lib/ghtorrent/migrations`

File: lib/ghtorrent/command.rb (hunk 3 above)
Comment (gousiosg) [inline lib/ghtorrent/command.rb]: I think it makes sense. Just make sure you check it works :-)

**Comment by jeffmcaffer** [comment]
closing in favor of a revised request

**Pull request closed without merging**